### PR TITLE
ARUHA-2538: repartitioning itself combined

### PR DESCRIPTION
**Pull request by adyach**
# One-line summary
Combine depended tasks interface to repartition event type

Add partitions to active timeline Kafka topic
Update timelines of the event type to have offsets for partitions added
Update event type cache
Close streaming connections for the subscriptions

### CODE REVIEW

File: src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java [hunk 1]
```diff
@@ -172,6 +171,12 @@ public void updateTimeLineForRepartition(final EventType eventType, final int pa
             }
             timelineDbRepository.updateTimelime(timeline);
         }
+
+        try {
+            eventTypeCache.updated(eventType.getName());
+        } catch (Exception e) {
+            throw new NakadiBaseException(e.getMessage(), e);
```
Comment (lmontrieux) [inline src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java]: Is there a better exception we could throw?

File: src/main/java/org/zalando/nakadi/service/EventTypeService.java [hunk 2]
```diff
@@ -537,22 +543,36 @@ private void repartitionEventType(
...
+    public void updateSubscriptionsForRepartitioning(final EventType eventType, final int partitions)
+            throws NakadiBaseException {
+        final List<Subscription> subscriptions = subscriptionRepository.listSubscriptions(
+                ImmutableSet.of(eventType.getName()), Optional.empty(), 0, 1000);
+        for (final Subscription subscription : subscriptions) {
+            final ZkSubscriptionClient zkClient = subscriptionClientFactory.createClient(subscription,
+                    LogPathBuilder.build(subscription.getId(), "repartition"));
+            zkClient.repartitionTopology(eventType.getName(), partitions);
+            zkClient.closeSubscriptionStreams(
+                    () -> LOG.info("subscription streams were closed, after repartitioning"), 60000);
```
Comment (lmontrieux) [inline src/main/java/org/zalando/nakadi/service/EventTypeService.java]: Can we use a constant (and also in line 567?)

**Comment by lmontrieux** [comment]
@adyach I have finished the review

File: src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java (hunk 1 above)
Comment (adyach) [reply src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java]: I do not see a reason for that, at the end it is converted to internal server error in the calling method

File: src/main/java/org/zalando/nakadi/service/EventTypeService.java (hunk 2 above)
Comment (adyach) [reply src/main/java/org/zalando/nakadi/service/EventTypeService.java]: done

**Comment by lmontrieux** [comment]
:+1:

**Comment by adyach** [comment]
👍

**Comment by lmontrieux** [comment]
:+1:

**Pull request merged**